controllers/clusteroperator.go

```diff
@@ -0,0 +1,222 @@
+	defaultCO := &osconfigv1.ClusterOperator{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterOperator",
+			APIVersion: "config.openshift.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterOperatorName,
+		},
+		Status: osconfigv1.ClusterOperatorStatus{
+			Conditions:     defaultStatusConditions(),
+			RelatedObjects: relatedObjects(),
+		},
+	}
+	operatorv1helpers.SetOperandVersion(&defaultCO.Status.Versions, osconfigv1.OperandVersion{Name: "operator", Version: os.Getenv("RELEASE_VERSION")})
+
+	co, err := r.OSClient.ConfigV1().ClusterOperators().Create(context.Background(), defaultCO, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	co.Status = defaultCO.Status
+	return r.OSClient.ConfigV1().ClusterOperators().UpdateStatus(context.Background(), co, metav1.UpdateOptions{})
+}
+
+// getOrCreateClusterOperator gets the existing CO, failing which it creates a new CO.
+func (r *ProvisioningReconciler) getOrCreateClusterOperator() (*osconfigv1.ClusterOperator, error) {
+	existing, err := r.OSClient.ConfigV1().ClusterOperators().Get(context.Background(), clusterOperatorName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		glog.Errorf("cluster baremetal operator does not exist, creating a new one.")
+		return r.createClusterOperator()
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get clusterOperator %q: %v", clusterOperatorName, err)
+	}
+
+	return existing, nil
+}
+
+// setStatusCondition initalizes and returns a ClusterOperatorStatusCondition
+func setStatusCondition(conditionType osconfigv1.ClusterStatusConditionType,
+	conditionStatus osconfigv1.ConditionStatus, reason string,
+	message string) osconfigv1.ClusterOperatorStatusCondition {
+	return osconfigv1.ClusterOperatorStatusCondition{
+		Type:               conditionType,
+		Status:             conditionStatus,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            message,
+	}
+}
+
+// getOperandVersions returns current version
+func getOperandVersions() []osconfigv1.OperandVersion {
+	operandVersions := []osconfigv1.OperandVersion{}
+	if releaseVersion := os.Getenv("RELEASE_VERSION"); len(releaseVersion) > 0 {
+		operandVersions = append(operandVersions, osconfigv1.OperandVersion{Name: "operator", Version: releaseVersion})
+	}
+	return operandVersions
+}
+
+//syncStatus applies the new condition to the mao ClusterOperator object.
+func (r *ProvisioningReconciler) syncStatus(co *osconfigv1.ClusterOperator, conds []osconfigv1.ClusterOperatorStatusCondition) error {
+	for _, c := range conds {
+		v1helpers.SetStatusCondition(&co.Status.Conditions, c)
+	}
+
+	_, err := r.OSClient.ConfigV1().ClusterOperators().UpdateStatus(context.Background(), co, metav1.UpdateOptions{})
+	return err
+}
+
+// updateCOStatusProgressing updates the ClusterOperator's status to Progressing
+func (r *ProvisioningReconciler) updateCOStatusProgressing() error {
```

Comment:
These snuck back into the latest commit after I split the commits a short while ago. I'll remove this since we are not using it in reconcile at this time.